### PR TITLE
Add new utility fill_fraction

### PR DESCRIPTION
**Pull request by micaeljtoliveira**
Add new utility to set depth to zero of cells with a sea area fraction smaller than some user-provided value.

### CODE REVIEW

**Comment by micaeljtoliveira** [comment]
@aekiss Would you mind having a look to see if this makes sense? The idea would be to call this before generating the mask, as discussed yesterday.

File: src/topogtools.f90 [hunk 1]
```diff
@@ -173,6 +178,17 @@ program topogtools
     call topog%update_history(get_mycommand())
     call topog%write(file_out)
 
+  case ('fill_fraction')
+    sea_area_fraction = rget('fraction')
+    if (sea_area_fraction <= 0.0 .and. sea_area_fraction >= 1.0) then
```
Comment (aekiss) [inline src/topogtools.f90]: replace `.and.` with `.or.`

File: src/topography.f90 [hunk 2]
```diff
@@ -443,6 +444,29 @@ subroutine topography_min_max_depth(this, vgrid, level)
...
+
+    if (any(this%frac < sea_area_fraction)) then
+      where (this%frac < sea_area_fraction)
+        this%depth = 0.0
+        this%frac = 0.0
+      end where
+
+      if (this%lakes_removed == 'yes') then
+        ! We might have created new lakes, so rerun deseas
+        call this%deseas()
```
Comment (aekiss) [inline src/topography.f90]: I don't have strong feelings either way, but would it be more intuitive (ie maintain orthogonality between these operations) to omit `call this%deseas()` and instead set `lakes_removed = "no "` and add a warning in the help that lakes can be created so deseas should be re-run?

**Review by aekiss — DISMISSED** [review]
Thanks, looks good, just a couple of comments

File: src/topography.f90 (hunk 2 above)
Comment (micaeljtoliveira) [reply src/topography.f90]: I was unsure about this, also when I implemented a similar logic in `fix_nonadvective`). I went for the option that would prevent mistakes (like someone forgetting to run deseas after `fix_nonadvective` or `fill_fraction`), but maybe it's better to assume users know what they are doing and only do what exactly what they requested. It also helps understanding what is exactly happening if all the operations are explicitly listed in the script that creates and fixes the topography.

File: src/topogtools.f90 (hunk 1 above)
Comment (micaeljtoliveira) [reply src/topogtools.f90]: Good catch!

File: src/topography.f90 (hunk 2 above)
Comment (micaeljtoliveira) [reply src/topography.f90]: I changed this. I will change the behavior of `fix_nonadvective` in a new PR, as I also want to include a few more changes.

File: src/topography.f90 [hunk 3]
```diff
@@ -443,6 +444,29 @@ subroutine topography_min_max_depth(this, vgrid, level)
...
+
+    if (any(this%frac < sea_area_fraction)) then
+      where (this%frac < sea_area_fraction)
+        this%depth = 0.0
+        this%frac = 0.0
+      end where
+
+      if (this%lakes_removed == 'yes') then
+        ! We might have created new lakes, so rerun deseas
+        this%lakes_removed = 'no'
```
Comment (aekiss) [inline src/topography.f90:462]: should this be 'no ' (3 chars)?

File: src/topography.f90 [hunk 4]
```diff
@@ -443,6 +444,29 @@ subroutine topography_min_max_depth(this, vgrid, level)
...
+    write(output_unit,'(a,f7.2)') "Filling cells that have a sea area fraction smaller than ", sea_area_fraction
+
+    if (any(this%frac < sea_area_fraction)) then
+      where (this%frac < sea_area_fraction)
+        this%depth = 0.0
+        this%frac = 0.0
+      end where
+
+      if (this%lakes_removed == 'yes') then
+        ! We might have created new lakes, so rerun deseas
```
Comment (aekiss) [inline src/topography.f90]: clearer to say `! We might have created new lakes. To fix, rerun deseas.`

File: src/topography.f90 (hunk 3 above)
Comment (micaeljtoliveira) [reply src/topography.f90:462]: In a case like this, spaces are automatically added to match the variable length.

Comment (micaeljtoliveira) [reply src/topography.f90:462]: Note that this is different than `this%lakes_removed(1:2) = 'no'`, which would effectively leave the third character unchanged.

File: src/topography.f90 (hunk 4 above)
Comment (micaeljtoliveira) [reply src/topography.f90]: I changed the comment to `! We might have created new lakes, so update the corresponding attribute`, as this comment is intended to explain why we are changing the attribute.

**Pull request merged**